src/azure-cli-core/azure/cli/core/extension/__init__.py

```diff
@@ -97,29 +101,31 @@ def get_all():
 
 class WheelExtension(Extension):
 
-    def __init__(self, name):
-        super(WheelExtension, self).__init__(name, 'whl')
+    def __init__(self, name, path):
+        super(WheelExtension, self).__init__(name, 'whl', path)
 
     def get_version(self):
         return self.metadata.get('version')
 
     def get_metadata(self):
         from wheel.install import WHEEL_INFO_RE
+
         if not extension_exists(self.name):
             return None
         metadata = {}
-        ext_dir = get_extension_path(self.name)
+        ext_dir = self.path or get_extension_path(self.name)
         dist_info_dirs = [f for f in os.listdir(ext_dir) if f.endswith('.dist-info')]
```

Comment:
The flow currently in the docs, with the `pip install --upgrade --target <> <>` fits this criteria right?
They have dist-infos so they should show up as wheel extensions?

Reply:
not true:
```
$ pip install --upgrade --target ../.azure/cliextensions/storage-preview/ ./src/storage-preview/
Processing c:\users\wilx\azure-cli-extensions\src\storage-preview
Building wheels for collected packages: storage-preview
  Running setup.py bdist_wheel for storage-preview: started
  Running setup.py bdist_wheel for storage-preview: finished with status 'done'
  Stored in directory: C:\Users\wilx\AppData\Local\pip\Cache\wheels\e1\f5\27\cd0ef0e934cde9c7e636a1aaec07742af7b22ec79dba4624f6
Successfully built storage-preview
Installing collected packages: storage-preview
Successfully installed storage-preview-0.1.7
```
It uses wheels as the default

Reply:
Interesting, this is true for the storage extension. However, if you try other extensions (just tried aem and alias) you will find that this same syntax generates an egg-info directory, not a dist-info one.

Reply:
Well, it turns out that `azdev extension add` always produces an `egg-info` folder, whereas the `pip install` command in the extension docs could either produce a `dist-info` or and `egg-info`.

Reply:
Yep, it does however look like the recent versions of pip favor installation through wheel. However, per our discussion, we should rather differentiate the "dev" extensions from the main ones through the use of the extension_dev_source environment variable as opposed to the current extension_dir environment variable.